adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandSubscriptionsManager.java

```diff
@@ -70,18 +70,25 @@ public CommandSubscriptionsManager(final Vertx vertx, final T config) {
      * Invoked when a device sends an MQTT <em>PUBACK</em> packet.
      *
      * @param msgId The message/packet id of the command published with QoS 1.
+     * @return {@code true} if a pending command request for the PUBACK was found.
      * @throws NullPointerException if msgId is {@code null}.
      */
-    public void handlePubAck(final Integer msgId) {
+    public boolean handlePubAck(final Integer msgId) {
         Objects.requireNonNull(msgId);
-        LOG.trace("acknowledgement received for command sent to device [packet-id: {}]", msgId);
-        Optional.ofNullable(removeFromWaitingForAcknowledgement(msgId))
-                .ifPresentOrElse(pendingCommandRequest -> {
+        return Optional.ofNullable(removeFromWaitingForAcknowledgement(msgId))
+                .map(pendingCommandRequest -> {
+                    LOG.trace("acknowledgement received for command sent to device [packet-id: {}]", msgId);
                     if (pendingCommandRequest.timerId != null) {
                         cancelTimer(pendingCommandRequest.timerId);
                     }
                     pendingCommandRequest.onAckHandler.handle(msgId);
-                }, () -> LOG.debug("no active command request found for received acknowledgement [packet-id: {}]", msgId));
+                    return true;
+                }).orElseGet(() -> {
+                    if (!subscriptions.isEmpty()) {
```

Comment:
Is this check needed? It seems to me that this log message makes sense even when there are no subscriptions (i.e the _map_ is empty) .

Reply:
This is an extra check in preparation for the upcoming error topic handling (see also other comment).
A PUBACK received while `subscriptions` is empty is most certainly not a PUBACK for a command message (but instead for a published error message for example).